### ISSUE TITLE
syntax highlight && when used in RUN commands?

### ISSUE THREAD

**Issue by sandywater** [jagregory/language-docker#21]
I think it would make for some beautiful Dockerfiles.

Also I have forgotten an && or a \ at the end of a line only to have a 20 min build fail at the last second due to that.

maybe the regex would be `^&&` or `&& \`

**Comment by jagregory** [comment]
This would be a nice compliment to the `/` highlighting we just added. PRs welcome, otherwise I'll have a look in due course.

**Comment by mishak87** [comment]
It would make more sense to add shellcheck support instead, by using syntax shell for `[ON BUILD] RUN` commands.

**Comment by xamgore** [comment]
@jagregory @mishak87, what do you think about this one?

<img src="https://cloud.githubusercontent.com/assets/4586392/26719467/5123e182-478d-11e7-9fe8-975b8d288642.png" width="440">

I inlined the shellscript grammar into the `RUN <shell>` expression, and also made a support of `\`.

**Comment by jagregory** [comment]
I think that looks great!

**Comment by klippx** [comment]
@xamgore Could you also please add `syntax--operator` for the `&&`? I would love to syntax highlight them a bit differently than `echo` and `test` in the examples

**Comment by xamgore** [comment]
@klippx, there are two ways here. The first one it to make a pull request to the `language-shell` package. The second one is just to drop the inner `shell` scope I have made, in favor of self written implementation. I'm not sure what is the best one.

**Comment by klippx** [comment]
@xamgore I think flagging the thing as `syntax--shell` is the best solution, go for it! Did you already pull request it?

**Comment by xamgore** [comment]
@klippx not yet. I want to check whether docker accepts the command with newlines:

```
RUN for (( i = 0; i < 10; i++ )); do
      echo $i;
done
```

If it turns out that no, but Atom still hightlights it, then… I will need some time to fix this.

**Comment by mishak87** [comment]
@xamgore End of line slashes are docker proprietary format. They have nothing to do with shell slashes (command continues on next line). Docker simply join lines together.
```Dockerfile
RUN a \
b \
  c
```
Results in:
```
sh -c 'a_b___c'
```
*Note: I have replaced spaces with undescores to more clearly demonstrate what is happening*

**Comment by ariasuni** [comment]
Still no support for shell syntax highlighting in language-docker? That would be pretty cool.

**Comment by aslafy-z** [comment]
Agreed, I often use `#` comments in my shell pipelines and it constantly throws "Empty / bogus instruction is invalid".

```
RUN true \
# blabla this
  && do_this \
# blabla that
  && do_that
```
